tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -155,6 +155,24 @@ function testTypeCheckingAgainstEffectiveType2() {
     assertEquality("bar", farr[3]);
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+
+function testR1(R1 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR2(R2 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testTypeCheckingInUnions() {
```

Comment:
This name doesn't sufficiently convey what we're testing/what we fixed.

Reply:
Changed the name